panic on wallet restore

panic: runtime error: index out of range [-1]
--
  |  
  | goroutine 24 [running]:
  | github.com/planetdecred/godcr/ui/page/wallets.(*Restore).onSuggestionSeedsClicked(0xc003ad7c00)
  | /home/user/code/godcr/ui/page/wallets/restore_page.go:244 +0x375
  | github.com/planetdecred/godcr/ui/page/wallets.(*Restore).HandleUserInteractions(0xc003ad7c00)
  | /home/user/code/godcr/ui/page/wallets/restore_page.go:586 +0x7d2
  | github.com/planetdecred/godcr/ui.(*Window).displayWindow(0xc000246500, {{0xc094d8f508c63de8, 0x591d6d34c261, 0x27e2880}, {0x3f800000, 0x3f800000}, {0x77c, 0x411}, {{0x0, 0x0}, ...}, ...})
  | /home/user/code/godcr/ui/window.go:233 +0x91
  | github.com/planetdecred/godcr/ui.(*Window).HandleEvents(0xc000246500)
  | /home/user/code/godcr/ui/window.go:203 +0x10c
  | main.main.func2()
  | /home/user/code/godcr/main.go:80 +0x26
  | created by main.main
  | /home/user/code/godcr/main.go:79 +0x4d3

--- a/ui/page/wallets/restore_page.go
+++ b/ui/page/wallets/restore_page.go
@@ -239,17 +239,19 @@ func (pg *Restore) inputsGroup(gtx layout.Context, l *layout.List, len, startInd
 
 func (pg *Restore) onSuggestionSeedsClicked() {
 	index := pg.seedEditors.focusIndex
-	for i, b := range pg.seedMenu {
-		for pg.seedMenu[i].button.Clicked() {
-			pg.seedEditors.editors[index].Edit.Editor.SetText(b.text)
-			pg.seedEditors.editors[index].Edit.Editor.MoveCaret(len(b.text), 0)
-			pg.seedClicked = true
-			if index != numberOfSeeds {
-				pg.seedEditors.editors[index+1].Edit.Editor.Focus()
-			}
+	if index != -1 {
+		for i, b := range pg.seedMenu {
+			for pg.seedMenu[i].button.Clicked() {
+				pg.seedEditors.editors[index].Edit.Editor.SetText(b.text)
+				pg.seedEditors.editors[index].Edit.Editor.MoveCaret(len(b.text), 0)
+				pg.seedClicked = true
+				if index != numberOfSeeds {
+					pg.seedEditors.editors[index+1].Edit.Editor.Focus()
+				}
 
-			if index == numberOfSeeds {
-				pg.isLastEditor = true
+				if index == numberOfSeeds {
+					pg.isLastEditor = true
+				}
 			}
 		}
 	}

